src/modules/cacheClient/cacheClient.service.ts

```diff
@@ -87,14 +87,18 @@ export class CacheClient implements ICacheClient {
         return Promise.reject(error);
     }
 
-    async testLogin(): Promise<void> {
-        // Simple test to check if logged in or no. TODO: have dedicated endpoint on the cache-server
-        // If receive unauthorized response, expect that refreshToken() will be called
-        await this.getRoleDefinition("testing.if.logged.in");
-    }
-
     async login() {
-        await this.testLogin();
+        const pubKeyAndIdentityToken = await this._signerService.publicKeyAndIdentityToken();
```

Comment:
@JGiter Will this mean that the signature is required each user loads `IAM` (e.g. refreshes the page in SB)?

Reply:
Ah, indeed. Will fix this too. Will read it from local storage

Reply:
That is basically what `testLogin` was doing. It was forcing a request so as to be able to control the timing of the signature request.
```
async testLogin(): Promise<void> {
        // Simple test to check if logged in or no. TODO: have dedicated endpoint on the cache-server
        // If receive unauthorized response, expect that refreshToken() will be called
        await this.getRoleDefinition("testing.if.logged.in");
    }
```
The reason for calling it on `login` is that I think it is a better UX if the signature is required when they first load the app/page. Otherwise it could be weird to be asked to sign something at a random point.

Reply:
Make sense. will revert it